src/test/java/com/marizueva/laboratory/hw2/ex2/LoginTestWithSoftAssertions.java

```diff
@@ -0,0 +1,243 @@
+        WebElement waterLogTrue = driver.findElement(By
+                .xpath("//*[contains(text(),"
+                        + "substring('01:14:22  Water: condition changed to true',11, 16))]"));
+        softAssert.assertTrue(waterLogTrue.isDisplayed());
+        softAssert.assertEquals(waterLogTrue.getTagName(), "li");
+
+        WebElement windLogTrue = driver.findElement(By
+                .xpath("//*[contains(text(),"
+                        + "substring('009:38:12  Wind: condition changed to true',11, 15))]"));
+        softAssert.assertTrue(windLogTrue.isDisplayed());
+        softAssert.assertEquals(windLogTrue.getTagName(), "li");
+
+        // Select radio Selen
+        WebElement radioSelen = driver.findElement(By
+                .cssSelector("label:nth-child(4) > input[type=radio]"));
+        radioSelen.click();
+
+        // Assert that for radiobutton there is a log row and value is corresponded
+        // to the status of radiobutton.
+        // Log row is displayed, radiobutton name and its status is corresponding to selected
+        WebElement selenLog = driver.findElement(By
+                .xpath("//*[contains(text(), substring('10:40:57 metal: "
+                        + "value changed to  Selen',23,40))] "));
+        softAssert.assertTrue(selenLog.isDisplayed());
+        softAssert.assertEquals(selenLog.getText().substring(22),
+                "changed to Selen");
+
+        // Select in dropdown Yellow
+        WebElement openColorDropdown = driver.findElement(By
+                .cssSelector("div.colors>.uui-form-element"));
+        openColorDropdown.click();
+        WebElement dropdownYellow = driver.findElement(By
+                .xpath("//div[@class='colors']/select/option[text()='Yellow']"));
+        dropdownYellow.click();
+
+        // Assert that for dropdown there is a log row and value is corresponded
+        // to the selected value.
+        // Log row is displayed, dropdown name and selected value is corresponding to selected
+        WebElement dropdownYellowLog = driver.findElement(By.xpath(
+                "//*[contains(text(),substring('11:18:10 Colors: value"
+                        + " changed to Yellow',18, 40))]"));
+        softAssert.assertTrue(dropdownYellowLog.isDisplayed());
+        softAssert.assertEquals(dropdownYellowLog.getText().substring(17),
+                "value changed to Yellow");
+
+        // Unselect and assert checkboxes Water, Wind Elements are unchecked
+        waterCheckbox.click();
+        windCheckbox.click();
+
+
+        // Assert that for each checkbox there is an individual log row
+        // and value is corresponded to the status of checkbox.
+        // Log rows are displayed, checkbox name and its status is corresponding to selected
+        WebElement waterLogFalse = driver.findElement(By.xpath(
+                "//*[contains(text(),"
+                        + "substring('11:23:22  Water: condition changed to false',11, 43))]"));
```

Comment:
Looks like that similar comments I've already put in the hw3 review. could you please replace locator without using text